testgen/testdata_gen.py

```diff
@@ -714,7 +711,10 @@ def generateCollTestData2(filename,
             continue
 
         is_compare = compare_pattern.match(line_in)
+        compare_type = None
         if is_compare:
+            # Reset string1 to the empty string.
```

Comment:
```suggestion
            # Initialize string1 to the empty string.
```

Reply:
Looks like this is the initialization before we begin the while loop, and we only update `string1` at the end of the loop body.